### PR TITLE
pmix/pmix112: Fix some int vs. pmix_status_t issues for XL compiler

### PR DESCRIPTION
**Pull request by jjhursey**

### CODE REVIEW

File: opal/mca/pmix/pmix112/pmix/src/buffer_ops/unpack.c [hunk 1]
```diff
@@ -967,7 +968,8 @@ pmix_status_t pmix_bfrop_unpack_topo(pmix_buffer_t *buffer, void *dest,
 {
     /* NOTE: hwloc defines topology_t as a pointer to a struct! */
     hwloc_topology_t t, *tarray  = (hwloc_topology_t*)dest;
-    pmix_status_t rc=PMIX_SUCCESS, i, cnt, j;
+    pmix_status_t rc=PMIX_SUCCESS, i, j;
```
Comment (rhc54) [inline opal/mca/pmix/pmix112/pmix/src/buffer_ops/unpack.c]: Shouldn't i and j be ints?

**Comment by rhc54** [comment]
Looks good, with that one comment

File: opal/mca/pmix/pmix112/pmix/src/buffer_ops/unpack.c (hunk 1 above)
Comment (jjhursey) [inline opal/mca/pmix/pmix112/pmix/src/buffer_ops/unpack.c]: Yeah, they should be int32_t as well. I'll make the change and push an update in a couple of minutes. Thanks.

**Comment by PHHargrove** [comment]
Works for me with both XLC 13.1 and 12.1  :+1:

**Comment by jjhursey** [comment]
@rhc54 just pushed the update per your comment. I think it is ready to merge.

File: opal/mca/pmix/pmix112/pmix/src/buffer_ops/open_close.c [hunk 2]
```diff
@@ -39,8 +40,8 @@
  * globals
  */
 bool pmix_bfrop_initialized = false;
-int pmix_bfrop_initial_size = 0;
-int pmix_bfrop_threshold_size = 0;
+pmix_status_t pmix_bfrop_initial_size = 0;
+pmix_status_t pmix_bfrop_threshold_size = 0;
```
Comment (rhc54) [inline opal/mca/pmix/pmix112/pmix/src/buffer_ops/open_close.c]: Uh - these don't look right. Those aren't functions but rather size variables - if anything, they maybe should be size_t. I'll fix this after I absorb the PR.

Comment (jjhursey) [inline opal/mca/pmix/pmix112/pmix/src/buffer_ops/open_close.c]: I made these match their definitions in `internal.h`, but I think your are right here I probably should have updated the `internal.h` to be `size_t`. Sorry about that, I used the header as authority in the changes and probably should have looked closer here.

**Pull request merged**